Marlin/src/pins/stm32f1/pins_FLSUN_HISPEED.h

```diff
@@ -109,88 +109,73 @@
 //
 // Drivers
 //
+/**
+ * This board has no hard-wired UART pins for TMC drivers.
+ * Several wiring options are provided below, defaulting to
+ * to the most compatible.
+ */
 #if HAS_TMC220x
-
-  #if ENABLED(HARDWARE_SERIAL)  /*  TMC2209 */
-    #define X_SLAVE_ADDRESS                    3  // |  |  :
-    #define Y_SLAVE_ADDRESS                    2  // :  |  :
-    #define Z_SLAVE_ADDRESS                    1  // |  :  :
-    //#define E0_SLAVE_ADDRESS                 0  // :  :  :
-
-    #define X_SERIAL_TX_PIN                 PA9   // TXD1
-    #define X_SERIAL_RX_PIN                 PA9   // TXD1
-
-    #define Y_SERIAL_TX_PIN                 PA9   // TXD1
-    #define Y_SERIAL_RX_PIN                 PA9   // TXD1
-
-    #define Z_SERIAL_TX_PIN                 PA9   // TXD1
-    #define Z_SERIAL_RX_PIN                 PA9   // TXD1
-
-  #elif ENABLED(SOFTWARE_SERIAL)  /*  TMC220x   */
-    /**
-     * TMC2208 stepper UART-configurable by PDN_UART pin
-     * Software serial
-     */
-    #define X_SLAVE_ADDRESS                    0
-    #define Y_SLAVE_ADDRESS                    0
-    #define Z_SLAVE_ADDRESS                    0
-
-    #define X_SERIAL_TX_PIN                 PA10  // RXD1
-    #define X_SERIAL_RX_PIN                 PA10  // RXD1
-
-    #define Y_SERIAL_TX_PIN                 PA9   // TXD1
-    #define Y_SERIAL_RX_PIN                 PA9   // TXD1
-
-    #define Z_SERIAL_TX_PIN                 PC7   // IO1
-    #define Z_SERIAL_RX_PIN                 PC7   // IO1
-
-  #endif
-  // Reduce baud rate to improve software serial reliability
-  #define TMC_BAUD_RATE                    19200
+  // SoftwareSerial with one pin per driver
+  // Compatible with TMC2208 and TMC2209 drivers
+  #define X_SERIAL_TX_PIN                   PA10  // RXD1
+  #define X_SERIAL_RX_PIN                   PA10  // RXD1
+  #define Y_SERIAL_TX_PIN                   PA9   // TXD1
+  #define Y_SERIAL_RX_PIN                   PA9   // TXD1
+  #define Z_SERIAL_TX_PIN                   PC7   // IO1
+  #define Z_SERIAL_RX_PIN                   PC7   // IO1
+  #define TMC_BAUD_RATE                   19200
+
+  // HardwareSerial with one pins for four drivers
+  // Compatible with TMC2209. Provides best performance.
+  // Requires SLAVE_ADDRESS definitions in Configuration_adv.h
+  // and proper jumper configuration. Uses one I/O pins
+  // like PA10/PA9/PC7/PA8 only.
+  // position the jumpers in this way:  |= close and := open.
+  // ex:
+  //#define  X_SLAVE_ADDRESS 3    // |  |  :
+  //#define  Y_SLAVE_ADDRESS 2    // :  |  :
+  //#define  Z_SLAVE_ADDRESS 1    // |  :  :
+  // For E 0 by default whatever the mode used but remove all jumpers.  
+  
```

Comment:
Something isn't right in the latest update. You have this comment, but none of the code that goes with it...

Also, I think you have mixed up two of my comments into one.

>   // HardwareSerial with one pins for four drivers
>  // Compatible with TMC2209. Provides best performance.

This is for the version which defines MSerial1 instead of pins. Any mode which defines pins is SoftwareSerial, and is not "best performance".

Reply:
> Also, I think you have mixed up two of my comments into one.

Agree on the terms but when we validate an assembly of four stepper drivers (TMC2209) in UART linked to the same serial pin and we define their communication address by the position of the jumpers. Isn't that a hardware mode?

Reply:
SoftwareSerial = Uses any I/O pin, and serial bit timing is manually controlled by software. Timing is very inaccurate, so low baud rates help make it more reliable.

HardwareSerial = Uses UART peripheral in the MCU. Accurate bit timing, but only works on the correct RX/TX pins.
                             User does not specify pin numbers, just a serial port name (like `MSerial1`)

Slave addresses can be used with TMC2209 drivers with both SoftwareSerial or HardwareSerial.

Reply:
Aaah, Ok!! Thank you, it's clearer now, sorry for the inconvenience.